src/js/components/AlertDialogComponent.jsx

```diff
@@ -0,0 +1,72 @@
+var React = require("react/addons");
+var classNames = require("classnames");
+
+var Util = require("../helpers/Util");
+var ModalComponent = require("../components/ModalComponent");
+
+var AlertModalComponent = React.createClass({
+  displayName: "AlertModalComponent",
+
+  propTypes: {
+    data: React.PropTypes.shape({
+      actionButtonLabel: React.PropTypes.string.isRequired,
+      message: React.PropTypes.string.isRequired,
+      state: React.PropTypes.string.isRequired,
+      title: React.PropTypes.string.isRequired
+    }),
+    onAccept: React.PropTypes.func,
+    onDismiss: React.PropTypes.func
+  },
+
+  getDefaultProps: function () {
+    return {
+      data: null,
+      onAccept: Util.noop,
+      onDismiss: Util.noop
+    }
+  },
+
+  componentDidMount: function () {
+    React.findDOMNode(this.refs.button).focus();
+  },
+
+  handleAccept: function () {
+    this.props.onAccept();
+  },
+
+  handleDismiss: function () {
+    this.props.onDismiss();
+  },
+
+  render: function () {
+    var data = this.props.data;
+    var className = classNames("dialog", data.state);
+
+    return (
+      <ModalComponent
+          centered={true}
+          className={className}
+          dismissOnClickOutside={true}
+          ref="modalComponent"
+          onDestroy={this.handleDismiss}>
+        <div className="modal-header">
+          {data.title}
```

Comment:
If a property is not required then you should always perform null checking – not ideal. In this case I'd suggest `this.props.data` to be `isRequired` as we can't do much without it.